htdocs/core/class/html.formai.class.php

```diff
@@ -131,112 +134,130 @@ public function getSectionForAIPrompt($function = 'textgeneration', $format = ''
 					var instructions = $('#ai_instructions".$htmlContent."').val();
 					var timeoutfinished = 0;
 					var apicallfinished = 0;
+					htmlname = '".$htmlContent."';
+					format = '".dol_escape_js($format)."';
+					functionai = '".dol_escape_js($function)."';
 
 					$('#ai_status_message".$htmlContent."').show();
 					$('.icon-container .loader').show();
 					setTimeout(function() {
 						timeoutfinished = 1;
-						if (apicallfinished) {
-							$('#ai_status_message".$htmlContent."').hide();
-						}
-					}, 2000);
-
-					if ('".$function."' === 'imagegeneration') {
-						// Handle image generation request
-						$.ajax({
-							url: '". DOL_URL_ROOT."/ai/ajax/generate_content.php?token=".currentToken()."',
-							type: 'POST',
-							contentType: 'application/json',
-							data: JSON.stringify({
-								'format': '".dol_escape_js($format)."',			/* the format for output */
-								'function': '".dol_escape_js($function)."',		/* the AI feature to call */
-								'instructions': instructions,					/* the prompt string */
-							}),
-							success: function(response) {
-								console.log('Received image URL: '+response);
-
-	                            // make substitutions
-	                            let substit = ". json_encode($this->substit).";
-	                            for (let key in substit) {
-	                                if (substit.hasOwnProperty(key)) {
-	                                    // Replace the placeholder with its corresponding value
-	                                    response = response.replace(key, substit[key]);
-	                                }
-	                            }
-
-								// Assuming response is the URL of the generated image
-								var imageUrl = response;
-								$('#ai_image_result').html('<img src=\"' + imageUrl + '\" alt=\"Generated Image\" />');
-
-								// Clear the input field
-								$('#ai_instructions').val('');
-
-								apicallfinished = 1;
-								if (timeoutfinished) {
-									$('#ai_status_message').hide();
-								}
-							},
-							error: function(xhr, status, error) {
-								alert(error);
-								console.error('error ajax', status, error);
-								$('#ai_status_message').hide();
+						$('#ai_status_message".$htmlContent."').hide();
+					}, 30000);
```

Comment:
In a next PR, i recommand to move this value 30000 into a parameter "Time out delay for AI response" in the first tab of setup of AI.